packages/medusa/src/services/order-edit.ts

```diff
@@ -388,6 +411,85 @@ export default class OrderEditService extends TransactionBaseService {
     return orderEdit
   }
 
+  async addLineItem(orderEditId: string, data: AddOrderEditLineItemInput) {
+    return await this.atomicPhase_(async (manager) => {
+      const lineItemServiceTx = this.lineItemService_.withTransaction(manager)
+
+      const orderEdit = await this.retrieve(orderEditId, {
+        relations: [
+          "order",
+
+          "order.cart",
+          "order.cart.customer",
+          "order.cart.discounts",
+          "order.cart.discounts.rule",
+          "order.cart.gift_cards",
+          "order.cart.region",
+          "order.cart.region.tax_rates",
+          "order.cart.shipping_address",
+          "order.cart.shipping_methods",
+
+          "order.region",
+        ],
+      })
+
+      if (!OrderEditService.isOrderEditActive(orderEdit)) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          `Can not add an item to the edit with status ${orderEdit.status}`
+        )
+      }
+
+      const regionId = orderEdit.order.region_id
+
+      // 0. check inventory
+
+      await this.inventoryService_
+        .withTransaction(manager)
+        .confirmInventory(data.variant_id, data.quantity)
+
+      // 1. generate new line item from data
+
+      const newItem = await lineItemServiceTx.generate(
+        data.variant_id,
+        regionId,
+        data.quantity,
+        {
+          metadata: data.metadata,
+        }
+      )
+
+      const { id: lineItemId } = await lineItemServiceTx.create(newItem)
+      const lineItem = await lineItemServiceTx.retrieve(lineItemId)
+
+      // 2. generate line item adjustments
+
+      await this.lineItemAdjustmentService_
+        .withTransaction(manager)
+        .createAdjustments(orderEdit.order.cart, lineItem)
+
+      // 3. generate tax lines
+
+      const calcContext = await this.totalsService_
+        .withTransaction(manager)
+        .getCalculationContext(orderEdit.order)
```

Comment:
todo: the calculation context is missing the allocation on the newly created line item since it does not exists on the order yet. also the context could be computed only for the new line item

Reply:
yes good catch, I added your fix from the comment below for now 👍